purchases-js-hybrid-mappings/src/mappers/customer_info_mapper.ts

```diff
@@ -0,0 +1,125 @@
+    "subscriptionsByProductIdentifier": mapSubscriptionInfos(customerInfo.subscriptionsByProductIdentifier),
+  };
+}
+
+function mapEntitlementInfos(entitlementInfos: EntitlementInfos): Record<string, unknown> {
+  return {
+    "all": Object.fromEntries(
+      Object.entries(entitlementInfos.all).map(([key, value]) => [key, mapEntitlementInfo(value)])
+    ),
+    "active": Object.fromEntries(
+      Object.entries(entitlementInfos.active).map(([key, value]) => [key, mapEntitlementInfo(value)])
+    ),
+    "verification": "NOT_REQUESTED" // TODO: Trusted entitlements not available in JS SDK
+  };
+}
+
+function mapEntitlementInfo(entitlementInfo: EntitlementInfo): Record<string, unknown> {
+  return {
+    "identifier": entitlementInfo.identifier,
+    "isActive": entitlementInfo.isActive,
+    "willRenew": entitlementInfo.willRenew,
+    "periodType": mapPeriodType(entitlementInfo.periodType),
+    "latestPurchaseDate": entitlementInfo.latestPurchaseDate ? entitlementInfo.latestPurchaseDate.toISOString() : null,
+    "latestPurchaseDateMillis": entitlementInfo.latestPurchaseDate ? entitlementInfo.latestPurchaseDate.getTime() : null,
+    "originalPurchaseDate": entitlementInfo.originalPurchaseDate ? entitlementInfo.originalPurchaseDate.toISOString() : null,
+    "originalPurchaseDateMillis": entitlementInfo.originalPurchaseDate ? entitlementInfo.originalPurchaseDate.getTime() : null,
+    "expirationDate": entitlementInfo.expirationDate ? entitlementInfo.expirationDate.toISOString() : null,
+    "expirationDateMillis": entitlementInfo.expirationDate ? entitlementInfo.expirationDate.getTime() : null,
+    "store": entitlementInfo.store,
+    "productIdentifier": entitlementInfo.productIdentifier,
+    "productPlanIdentifier": entitlementInfo.productPlanIdentifier,
+    "isSandbox": entitlementInfo.isSandbox,
+    "unsubscribeDetectedAt": entitlementInfo.unsubscribeDetectedAt ? entitlementInfo.unsubscribeDetectedAt.toISOString() : null,
+    "unsubscribeDetectedAtMillis": entitlementInfo.unsubscribeDetectedAt ? entitlementInfo.unsubscribeDetectedAt.getTime() : null,
+    "billingIssueDetectedAt": entitlementInfo.billingIssueDetectedAt ? entitlementInfo.billingIssueDetectedAt.toISOString() : null,
+    "billingIssueDetectedAtMillis": entitlementInfo.billingIssueDetectedAt ? entitlementInfo.billingIssueDetectedAt.getTime() : null,
+    "ownershipType": entitlementInfo.ownershipType,
+    "verification": "NOT_REQUESTED", // TODO: Trusted entitlements not available in JS SDK
+  };
+}
+
+function mapNonSubscriptionTransactions(nonSubscriptionTransactions: NonSubscriptionTransaction[]): Record<string, unknown>[] {
+  return nonSubscriptionTransactions.map((transaction) => {
+    return {
+      "transactionIdentifier": transaction.transactionIdentifier,
+      "revenueCatId": transaction.transactionIdentifier, // Deprecated: Use transactionIdentifier in this map instead
```

Comment:
Kept these to keep the same interface as the mobile PHC libraries... I think it's probably not needed, but haven't checked all hybrids.